### PR TITLE
feat(projects): list all projects

### PR DESCRIPTION
**Pull request by betty2310**
## Related Tickets
- [ticket redmine](https://edu-redmine.sun-asterisk.vn/issues/67226)

## WHAT (optional)
- `/projects` wil list all projects
- support pagy
- can filter by name, group and status

## Evidence (Screenshot or Video)
CleanShot 2023-08-31 at 15 39 34

### CODE REVIEW

**Comment by betty2310** [comment]
ready

File: app/views/projects/_search_form.html.erb [hunk 1]
```diff
@@ -0,0 +1,20 @@
+<%= form_with url: projects_path, method: :get, class: "row g-4 pb-4" do |f| %>
+  <div class="col-auto">
+    <%= f.label :name,  t(".name") %>
+    <%= f.text_field :name, value: params[:name] %>
+  </div>
+  <div class="col-auto">
+    <%= f.label :group, t(".group") %>
+    <%= f.select :group, list_group, {include_blank: true, selected: @selected_group} %>
```
Comment (ngocvt-0484) [inline app/views/projects/_search_form.html.erb]: ở view gọi thẳng params[:group_id] k cần đặt biến

File: app/views/projects/_search_form.html.erb [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+  </div>
+  <div class="col-auto">
+    <%= f.label :status, t(".status") %>
+    <%= f.select :status, list_status, {include_blank: true, selected: @selected_status} %>
```
Comment (ngocvt-0484) [inline app/views/projects/_search_form.html.erb]: same

File: app/helpers/projects_helper.rb [hunk 3]
```diff
@@ -0,0 +1,15 @@
+module ProjectsHelper
+  def group_name project
+    project.group.name
```
Comment (ngocvt-0484) [inline app/helpers/projects_helper.rb]: dùng delegate rồi gọi thẳng ở view. k cần viết helper

**Comment by betty2310** [comment]
ready

**Pull request merged**